bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,314 @@
+import org.openhab.binding.salus.internal.rest.ApiResponse;
+import org.openhab.binding.salus.internal.rest.Device;
+import org.openhab.binding.salus.internal.rest.DeviceProperty;
+import org.openhab.binding.salus.internal.rest.Error;
+import org.openhab.binding.salus.internal.rest.GsonMapper;
+import org.openhab.binding.salus.internal.rest.HttpClient;
+import org.openhab.binding.salus.internal.rest.RestClient;
+import org.openhab.binding.salus.internal.rest.RetryHttpClient;
+import org.openhab.binding.salus.internal.rest.SalusApi;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public final class CloudBridgeHandler extends BaseBridgeHandler implements CloudApi {
+    private Logger logger = LoggerFactory.getLogger(CloudBridgeHandler.class.getName());
+    private final HttpClientFactory httpClientFactory;
+    @NonNullByDefault({})
+    private LoadingCache<String, SortedSet<DeviceProperty<?>>> devicePropertiesCache;
+    @Nullable
+    private SalusApi salusApi;
+    @Nullable
+    private ScheduledFuture<?> scheduledFuture;
+
+    public CloudBridgeHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
+        super(bridge);
+        this.httpClientFactory = httpClientFactory;
+    }
+
+    @Override
+    public void initialize() {
+        CloudBridgeConfig config = this.getConfigAs(CloudBridgeConfig.class);
+        if (!config.isValid()) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "@text/cloud-bridge-handler.initialize.username-pass-not-valid");
+            return;
+        }
+        RestClient httpClient = new HttpClient(httpClientFactory.getCommonHttpClient());
+        if (config.getMaxHttpRetries() > 0) {
+            httpClient = new RetryHttpClient(httpClient, config.getMaxHttpRetries());
+        }
+        @Nullable
+        SalusApi localSalusApi = salusApi = new SalusApi(config.getUsername(), config.getPassword().toCharArray(),
+                config.getUrl(), httpClient, GsonMapper.INSTANCE);
+        logger = LoggerFactory
+                .getLogger(CloudBridgeHandler.class.getName() + "[" + config.getUsername().replace(".", "_") + "]");
+
+        ScheduledExecutorService scheduledPool = ThreadPoolManager.getScheduledPool(SalusBindingConstants.BINDING_ID);
+        scheduledPool.schedule(() -> tryConnectToCloud(localSalusApi), 1, MICROSECONDS);
+
+        this.devicePropertiesCache = Caffeine.newBuilder().maximumSize(10_000)
```

Comment:
See other comments about exception handing. From the Salus API request/response you can probably tell if it is a timeout, authentication error or otherwise. At that place you can throw custom excpetion like `SalusAPIAuthenticationException` that can be catched here to give clear insight to the user on the ThingStatusDetail.
There should not be a need to catch `Exception`

Reply:
Do you want me to catch InterruptedException, TimeoutException, ExecutionException from [Request.send()](https://eclipse.dev/jetty/javadoc/jetty-10/org/eclipse/jetty/client/api/Request.html#send()) and then throw SalusCommunicationException (that extends RuntimeException?)

Reply:
Yes, that is a common pattern, becasue InterruptedException, TimeoutException, ExecutionException are specific to the API.

Reply:
Two more questions: 

1. How can I know that `send()` method is not throwing any other exception that is not listed?
2. How should I handle that `send()` method might change in the future and start throwing some other exception (even not listed ones)? In the future when someone will update the dependencies he will not look on all bindings for such a cases

Reply:
Look at the implementation of send method: https://github.com/jetty/jetty.project/blob/jetty-12.0.x/jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpRequest.java#L702

It can throw anything

Reply:
Exactly that is why you need to handle the exceptions within the class that actually sends the http requests, in the case for salus it was modelled in the `HttpClient` class. Some random examples of a equivalant classes in other bindings, but there are many more:
[automowerconnect](https://github.com/openhab/openhab-addons/blob/4fbe464a513ff638c3a1ea649bbec2c12fcaf4f2/bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/rest/api/automowerconnect/AutomowerConnectApi.java#L91)
[plugwiseha](https://github.com/openhab/openhab-addons/blob/4fbe464a513ff638c3a1ea649bbec2c12fcaf4f2/bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/api/model/PlugwiseHAControllerRequest.java#L247)
[enphase](https://github.com/openhab/openhab-addons/blob/4fbe464a513ff638c3a1ea649bbec2c12fcaf4f2/bundles/org.openhab.binding.enphase/src/main/java/org/openhab/binding/enphase/internal/handler/EnvoyConnector.java#L247)

Reply:
Ah! OK, now it's clear!